addons/point_of_sale/static/src/js/models.js

```diff
@@ -845,7 +873,87 @@ exports.PosModel = Backbone.Model.extend({
             this.get('orders').add(orders);
         }
     },
-
+    async _loadMissingProducts(orders) {
+        const missingProductIds = new Set([]);
+        for (const order of orders) {
+            for (const line of order.lines) {
+                const productId = line[2].product_id;
+                if (missingProductIds.has(productId)) continue;
+                if (!this.db.get_product_by_id(productId)) {
+                    missingProductIds.add(productId);
+                }
+            }
+        }
+        const productModel = _.find(this.models, function(model){return model.model === 'product.product';});
+        const fields = productModel.fields;
+        const products = await this.rpc({
+            model: 'product.product',
+            method: 'read',
+            args: [[...missingProductIds], fields],
+            context: Object.assign(this.session.user_context, { display_default_code: false }),
+        });
+        productModel.loaded(this, products);
+    },
+    async _loadMissingPartners(orders) {
+        const missingPartnerIds = new Set([]);
+        for (const order of orders) {
+            const partnerId = order.partner_id;
+            if(missingPartnerIds.has(partnerId)) continue;
+            if (partnerId && !this.db.get_partner_by_id(partnerId)) {
+                missingPartnerIds.add(partnerId);
+            }
+        }
+        const partnerModel = _.find(this.models, function(model){return model.model === 'res.partner';});
+        const fields = partnerModel.fields;
+        if(missingPartnerIds) {
+            const partners = await this.rpc({
+                model: 'res.partner',
+                method: 'read',
+                args: [[...missingPartnerIds], fields],
+                context: Object.assign(this.session.user_context, { display_default_code: false }),
+            });
+            partnerModel.loaded(this, partners);
+        }
+    },
+    loadProductsBackground: async function() {
+        this.env.pos.config.productsLoading = true;
+        let i = 1;
+        let ProductIds = [];
+        do {
+            ProductIds = await this.rpc({
+                model: 'product.product',
+                method: 'search',
+                args: [[]],
+                kwargs: {
+                    limit: this.env.pos.config.limited_products_amount,
+                    offset: this.env.pos.config.limited_products_amount * i
+                },
+                context: this.env.session.user_context,
+            });
+            this.env.pos._addProducts(ProductIds);
+            i += 1;
+        } while(ProductIds.length);
+    },
+    loadPartnersBackground: async function() {
+        this.env.pos.config.partnersLoading = true;
+        let i = 1;
+        let PartnerIds = [];
+        var fields = _.find(this.env.pos.models, function(model){ return model.label === 'load_partners'; }).fields;
```

Comment:
find comme ca:  _.find(this.models, (model) => model.model === 'product.product'); avec res.partner